### PR TITLE
feat(TosaToLinalg): use linalg.matmul when converting tosa.matmul with 1x batch.

### PR DESCRIPTION
**Pull request by ttjost**

### CODE REVIEW

File: mlir/lib/Conversion/TosaToLinalg/TosaToLinalgNamed.cpp [hunk 1]
```diff
@@ -525,20 +557,53 @@ class MatMulConverter : public OpConversionPattern<tosa::MatMulOp> {
       dynDims[2] = rewriter.create<tensor::DimOp>(loc, op->getOperand(1), 2);
     }
 
+    auto getTypeWithoutBatch = [&](ShapedType ty) {
```
Comment (mgehre-amd) [inline mlir/lib/Conversion/TosaToLinalg/TosaToLinalgNamed.cpp:560]: Will this work when dims are dynamic?

File: mlir/lib/Conversion/TosaToLinalg/TosaToLinalgNamed.cpp [hunk 2]
```diff
@@ -525,20 +557,53 @@ class MatMulConverter : public OpConversionPattern<tosa::MatMulOp> {
...
+      auto shape2D = {ty.getDimSize(1), ty.getDimSize(2)};
+      return RankedTensorType::get(shape2D, ty.getElementType());
+    };
+
     SmallVector<Value> filteredDims = condenseValues(dynDims);
 
+    bool isBatchOneDim =
+        outputTy.getDimSize(0) == 1 && this->useMatmulForSingleBatch;
```
Comment (mgehre-amd) [inline mlir/lib/Conversion/TosaToLinalg/TosaToLinalgNamed.cpp:568]: nit: useMatmulForSingleBatch doesn't really relate to whether `isBatchOneDim`

File: mlir/lib/Conversion/TosaToLinalg/TosaToLinalgNamed.cpp (hunk 1 above)
Comment (ttjost) [reply mlir/lib/Conversion/TosaToLinalg/TosaToLinalgNamed.cpp:560]: It does work, as long as `getReassociationIndicesForReshape` can resolve the indices. I have tests for that.

File: mlir/lib/Conversion/TosaToLinalg/TosaToLinalgNamed.cpp (hunk 2 above)
Comment (ttjost) [reply mlir/lib/Conversion/TosaToLinalg/TosaToLinalgNamed.cpp:568]: I'll rename it to a local `useMatmul`.

Comment (ttjost) [reply mlir/lib/Conversion/TosaToLinalg/TosaToLinalgNamed.cpp:568]: Done!

**Review by mgehre-amd — APPROVED** [review]

**Pull request merged**